website/blog/2024-03-07-using-inner-blocks.md

```diff
@@ -0,0 +1,162 @@
+				orientation='vertical'
+				renderAppender={() => <BlockInserter clientId={clientId} hasLabel additionalClasses='es-mb-6' />}
+			/>
+		</div>
+
+	);
+};
+```
+
+The `InnerBlocks` element is the default one from the Gutenberg editor. In order to limit the allowed blocks, we need to pass the attribute that contains the list of allowed inner blocks. The `orientation` prop determines if the inner blocks will be displayed horizontally or vertically in the editor.
+
+For the `renderAppender` prop, a custom `BlockInserter` element is used. This is a replacement for the default Gutenberg inserter, whose styling can vary between WP versions.
+
+Finally, to render the inner blocks in PHP, you need to add the following:
+
+```php
+<?php
+// phpcs:ignore Eightshift.Security.ComponentsEscape.OutputNotEscaped
+echo $innerBlockContent;
+?>
+```
+
+And that’s it! Your block now supports inner blocks.
+
+## Overriding inner block attributes
+
+Inner blocks behave like regular blocks, which means they also have a wrapper by default. This may not always be desired and manually removing wrapper or adding custom wrapper settings on each inner block can be tedious. That’s why Eightshift DevKit offers few helpers that allow you to override inner block attributes.
+
+Both of these helpers go to your `{block-name}-block.js` file.
+
+### overrideInnerBlockAttributes
+
+This helper allows you to set the attributes on all inner blocks by passing them as a third parameter. The example below shows how to disable the wrapper and hide wrapper controls.
+
+```jsx
+import { useSelect } from '@wordpress/data';
+import { overrideInnerBlockAttributes } from '@eightshift/frontend-libs/scripts';
+
+export const Accordion = (props) => {
+	// Set these attributes to all inner blocks once inserted in DOM.
+	useSelect((select) => {
+		overrideInnerBlockAttributes(
+			select,
+			props.clientId,
+			{
+				wrapperUse: false,
+				wrapperNoControls: true,
+			},
+		);
+	});
+
+	return (
+		// ...
+	);
+};
+```
+
+Keep in mind this helper allows you to set any attribute that the inner block supports, so you can even set a larger paragraph size and a different color, for example.
+
+### overrideInnerBlockSimpleWrapperAttributes
+
+The `overrideInnerBlockSimpleWrapperAttributes` sets up a simple wrapper to all inner blocks. This is similar to the helper method above, but already has predefined attributes for simple wrapper, so if you only want to set up simple wrapper, you don’t have to add the attributes manually.
+
+```jsx
+
+import { useSelect } from '@wordpress/data';
+import { overrideInnerBlockSimpleWrapperAttributes } from '@eightshift/frontend-libs/scripts';
+
+export const CardTwoColumn = (props) => {
+	const {
+		clientId,
+	} = props;
+
+	// Set these attributes to all inner blocks once inserted in DOM.
+	useSelect((select) => {
+		overrideInnerBlockSimpleWrapperAttributes(select, clientId);
+	});
+
+	return (
+		// ...
+	);
+};
+```
+
+## Conclusion
```

Comment:
Maybe before the conclusion you can also mention the newer "inline" innerblocks via `innerBlockProps` that allow more control over the InnerBlocks container.

```jsx
import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';

export const DemoEditor = (props) => {
	const blockProps = useBlockProps();
	const innerBlocksProps = useInnerBlocksProps(blockProps, {
		allowedBlocks: demoAllowedBlocks,
		renderAppender: () => <BlockInserter clientId={clientId} />,
	});

	return (
		<div {...innerBlocksProps} /> // Do whatever you want with this.
	);
};
```

Reply:
I've added this one, so feel free to update if I've missed anything or if it needs better explanation 🙂